### PR TITLE
Update and clean up several aspects of documentation tooling

### PR DESCRIPTION
**Pull request by Viicos**
Work towards https://github.com/pydantic/pydantic/issues/10083.

Remove `mkdocs-simple-hooks` dependency, this is now available in MkDocs
Remove `mkdocs-embed-external-markdown` and apply same logic in a hook instead. The extension was parsing every markdown file as Jinja2 templates, which had limitations with some syntax. There's also a small performance improvement in not doing so.
Update `griffe` to the latest version, fixing breaking changes to extension, and remove dependency to `backports-strenum`.
Do not unnecessarily depend on a git version of `mike`

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -104,21 +104,17 @@ substitutions = [
...
     "tomli",
     "pyupgrade",
-    "mike @ git+https://github.com/jimporter/mike.git@master",
-    "mkdocs-embed-external-markdown>=2.3.0",
+    "mike",
     "pytest-examples>=0.0.13",
     "pydantic-settings>=2.0b1",
     "pydantic-extra-types @ git+https://github.com/pydantic/pydantic-extra-types.git@main",
-    # Not available before v3.8.6, hence the pin here, as it's required by griffe
-    "backports-strenum==1.3.1; python_version >= '3.8.6' and python_version < '3.11'"
```
Comment (Viicos) [inline pyproject.toml:121]: This is one of the reasons I dislike the PDM behavior (as we discussed internally). We can end up with unused dependencies that have to be manually checked.

File: docs/plugins/main.py [hunk 2]
```diff
@@ -7,24 +7,32 @@
...
 logger = logging.getLogger('mkdocs.plugin')
 THIS_DIR = Path(__file__).parent
 DOCS_DIR = THIS_DIR.parent
 PROJECT_ROOT = DOCS_DIR.parent
 
 
+if TYPE_CHECKING:
+    from .conversion_table import conversion_table
+else:
+    from conversion_table import conversion_table
```
Comment (Viicos) [inline docs/plugins/main.py:31]: Not ideal. MkDocs does some `sys.path` magic so the absolute import works but type checkers don't understand it.

File: docs/plugins/main.py [hunk 3]
```diff
@@ -51,6 +59,8 @@ def on_page_markdown(markdown: str, page: Page, config: Config, files: Files) ->
         return md
     if md := render_why(markdown, page):
         return md
+    if md := render_pydantic_settings(markdown, page):
+        return md
```
Comment (Viicos) [inline docs/plugins/main.py:63]: Ideally we should move away from these pattern: it isn't really elegant and we could move to the [Blocks API](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/) at some point. It seems to provide similar functionality as Sphinx directives which are essential imo

File: docs/plugins/main.py [hunk 4]
```diff
@@ -10,21 +10,32 @@
...
 
 
+try:
+    from .conversion_table import conversion_table
+except ImportError:
+    # Due to how MkDocs requires this file to be specified (as a path and not a
+    # dot-separated module name), relative imports don't work:
+    # MkDocs is adding the dir. of this file to `sys.path` and uses
+    # `importlib.spec_from_file_location` and `module_from_spec`, which isn't ideal.
+    from conversion_table import conversion_table
```
Comment (Viicos) [inline docs/plugins/main.py:34]: If they had used `importlib.import_module`, things would have been easier I think.

File: pyproject.toml (hunk 1 above)
Comment (sydney-runkle) [reply pyproject.toml:121]: Yeah, I wasted some time on this a while ago and it was relatively... annoying

**Review by sydney-runkle — APPROVED** [review]
Looks good.

I don't think we can, but I wonder if we can handle `pydantic-settings` docs integration like we handle `pydantic-extra-types` - seems more simple. My hunch is no, given that we only use the api docs from `pydantic-extra-types`.

**Pull request merged**